[FEATURE] Sur le domaine pix.org mettre "en" comme fallback de locale sur tous les frontaux (PIX-17758)

lego-technix opened the pull request:
## 🌸 Problème

Sur le domaine `pix.org` c’est un public international qui est attendu et qui vient. Un utilisateur « international », indépendamment de sa langue maternelle, est majoritairement plus à l’aise avec l’anglais qu’avec le français. Or 
sur le domaine `pix.org` les frontaux sont par défaut affichés en français et il est nécessaire pour l’utilisateur de faire un changement de langue pour trouver une langue qui lui correspond ou à défaut l’anglais.

## 🌳 Proposition

Sur le domaine `pix.org`, et pas sur `pix.fr`, sur tous les frontaux (Pix App, Pix Orga, Pix Certif, Pix Admin), mettre la locale `en` comme fallback de locale, c’est à dire si aucune locale n’est spécifiée (soit dans les queryParam, soit dans comme locale de l’utilisateur).

Les commits de cette PR sont en 2 phases : 
1. Pour tous les frontaux définition et centralisation de la notion de `DEFAULT_LOCALE`,
2. Pour tous les frontaux ajout de la possibilité de surcharger la `DEFAULT_LOCALE` à partir d’une nouvelle variable d’environnement `DEFAULT_LOCALE`, de manière à ce que les tests, dont de grandes parties sont écrites pour le français au lieu d’être génériques ou plus génériques, continuent de fonctionner.

## 🐝 Remarques

Avec cette PR l’objectif n’est pas de rendre identique et factoriser en partie la gestion des locales entre les différents frontaux. Ce sujet fait l’objet d’une Epix dédiée. Dans cette PR on essaye néanmoins que le code des différents frontaux soit quand même le plus proche possible pour justement se pas s’éloigner de l’Epix en question.

## 🤧 Pour tester

1. Vérifier que les tests de la CI passent,
2. Sans être connecté, se rendre sur les frontaux Pix App, Pix Orga, Pix Certif et Pix Admin sur le domaine `pix.fr`,
3. Constater que ces frontaux présentent tous par défaut une interface utilisateur en français,
4. Sans être connecté, se rendre sur les frontaux Pix App, Pix Orga, Pix Certif et Pix Admin sur le domaine `pix.org`,
5. Constater que ces frontaux présentent tous par défaut une interface utilisateur en anglais. À noter que pour Pix Admin lorsqu’une traduction en anglais est absente, alors elle est remplacée par la traduction française.

lego-technix committed:
feat(mon-pix): set "en" as the default locale for .org

lego-technix committed:
feat(orga): set "en" as the default locale for .org

lego-technix committed:
feat(certif): set "en" as the default locale for .org

lego-technix committed:
feat(admin): set "en" as the default locale for .org but let translation fallback to "fr"

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr12375.review.pix.fr)
  * [App (.org)](https://app-pr12375.review.pix.org)
  * [Orga (.fr)](https://orga-pr12375.review.pix.fr)
  * [Orga (.org)](https://orga-pr12375.review.pix.org)
  * [Certif (.fr)](https://certif-pr12375.review.pix.fr)
  * [Certif (.org)](https://certif-pr12375.review.pix.org)
  * [Junior](https://junior-pr12375.review.pix.fr)
  * [Admin](https://admin-pr12375.review.pix.fr)
  * [API](https://api-pr12375.review.pix.fr/api/)
  * [API MaDDo](https://pix-api-maddo-review-pr12375.osc-fr1.scalingo.io/api/)
  * [Audit Logger](https://pix-audit-logger-review-pr12375.osc-fr1.scalingo.io/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr12375/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr12375/environment)
  * [scalingo api-maddo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-maddo-review-pr12375/environment)
  * [scalingo audit-logger](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr12375/environment)

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr12375.review.pix.fr)
  * [App (.org)](https://app-pr12375.review.pix.org)
  * [Orga (.fr)](https://orga-pr12375.review.pix.fr)
  * [Orga (.org)](https://orga-pr12375.review.pix.org)
  * [Certif (.fr)](https://certif-pr12375.review.pix.fr)
  * [Certif (.org)](https://certif-pr12375.review.pix.org)
  * [Junior](https://junior-pr12375.review.pix.fr)
  * [Admin](https://admin-pr12375.review.pix.fr)
  * [API](https://api-pr12375.review.pix.fr/api/)
  * [API MaDDo](https://pix-api-maddo-review-pr12375.osc-fr1.scalingo.io/api/)
  * [Audit Logger](https://pix-audit-logger-review-pr12375.osc-fr1.scalingo.io/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr12375/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr12375/environment)
  * [scalingo api-maddo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-maddo-review-pr12375/environment)
  * [scalingo audit-logger](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr12375/environment)

lego-technix committed:
refactor(mon-pix): get DEFAULT_LOCALE from ENV to be able to use different values in production and in tests

with DEFAULT_LOCALE='en' in production and DEFAULT_LOCALE='fr' for test.

This is needed because all the unit/integration/acceptance tests are not all
written with the translation keys, and because the e2e tests are written for the
"fr" locale.

lego-technix committed:
refactor(orga): get DEFAULT_LOCALE from ENV to be able to use different values in production and in tests

with DEFAULT_LOCALE='en' in production and DEFAULT_LOCALE='fr' for test.

This is needed because all the unit/integration/acceptance tests are not all
written with the translation keys, and because the e2e tests are written for the
"fr" locale.

lego-technix committed:
refactor(admin): get DEFAULT_LOCALE from ENV to be able to use different values in production and in tests

with DEFAULT_LOCALE='en' in production and DEFAULT_LOCALE='fr' for test.

This is needed because all the unit/integration/acceptance tests are not all
written with the translation keys.

lego-technix committed:
refactor(certif): get DEFAULT_LOCALE from ENV to be able to use different values in production and in tests

with DEFAULT_LOCALE='en' in production and DEFAULT_LOCALE='fr' for test.

This is needed because all the unit/integration/acceptance tests are not all
written with the translation keys.

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr12375.review.pix.fr)
  * [App (.org)](https://app-pr12375.review.pix.org)
  * [Orga (.fr)](https://orga-pr12375.review.pix.fr)
  * [Orga (.org)](https://orga-pr12375.review.pix.org)
  * [Certif (.fr)](https://certif-pr12375.review.pix.fr)
  * [Certif (.org)](https://certif-pr12375.review.pix.org)
  * [Junior](https://junior-pr12375.review.pix.fr)
  * [Admin](https://admin-pr12375.review.pix.fr)
  * [API](https://api-pr12375.review.pix.fr/api/)
  * [API MaDDo](https://pix-api-maddo-review-pr12375.osc-fr1.scalingo.io/api/)
  * [Audit Logger](https://pix-audit-logger-review-pr12375.osc-fr1.scalingo.io/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr12375/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr12375/environment)
  * [scalingo api-maddo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-maddo-review-pr12375/environment)
  * [scalingo audit-logger](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr12375/environment)

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr12375.review.pix.fr)
  * [App (.org)](https://app-pr12375.review.pix.org)
  * [Orga (.fr)](https://orga-pr12375.review.pix.fr)
  * [Orga (.org)](https://orga-pr12375.review.pix.org)
  * [Certif (.fr)](https://certif-pr12375.review.pix.fr)
  * [Certif (.org)](https://certif-pr12375.review.pix.org)
  * [Junior](https://junior-pr12375.review.pix.fr)
  * [Admin](https://admin-pr12375.review.pix.fr)
  * [API](https://api-pr12375.review.pix.fr/api/)
  * [API MaDDo](https://pix-api-maddo-review-pr12375.osc-fr1.scalingo.io/api/)
  * [Audit Logger](https://pix-audit-logger-review-pr12375.osc-fr1.scalingo.io/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr12375/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr12375/environment)
  * [scalingo api-maddo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-maddo-review-pr12375/environment)
  * [scalingo audit-logger](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr12375/environment)

theotime2005 reviewed (changes requested):
Pix admin ne doit pas être en anglais sur le domaine .fr

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr12375.review.pix.fr)
  * [App (.org)](https://app-pr12375.review.pix.org)
  * [Orga (.fr)](https://orga-pr12375.review.pix.fr)
  * [Orga (.org)](https://orga-pr12375.review.pix.org)
  * [Certif (.fr)](https://certif-pr12375.review.pix.fr)
  * [Certif (.org)](https://certif-pr12375.review.pix.org)
  * [Junior](https://junior-pr12375.review.pix.fr)
  * [Admin](https://admin-pr12375.review.pix.fr)
  * [API](https://api-pr12375.review.pix.fr/api/)
  * [API MaDDo](https://pix-api-maddo-review-pr12375.osc-fr1.scalingo.io/api/)
  * [Audit Logger](https://pix-audit-logger-review-pr12375.osc-fr1.scalingo.io/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr12375/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr12375/environment)
  * [scalingo api-maddo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-maddo-review-pr12375/environment)
  * [scalingo audit-logger](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr12375/environment)

lego-technix commented:
> Pix admin ne doit pas être en anglais sur le domaine .fr

Le problème sur Pix Admin sur le domaine `pix.fr` est corrigé. Vous pouvez reprendre vos revues 🙂

theotime2005 commented:
Est-ce vraiment judicieux d'autoriser l'utilisation de pix admin dans une autre langue à l'heure actuelle ?
En effet, beaucoup de chaînes ne sont pas traduite et l'internationalisation n'est pas prévu dans l'immédiat (je crois).

lego-technix commented:
> Est-ce vraiment judicieux d'autoriser l'utilisation de pix admin dans une autre langue à l'heure actuelle ? En effet, beaucoup de chaînes ne sont pas traduite et l'internationalisation n'est pas prévu dans l'immédiat (je crois).

Lors de l’Epix de _Remise à plat des langues & locales_ le code de tous les frontaux va être très factorisé. Aussi plus les frontaux ont les mêmes comportements mieux cela s’aligne avec la prochaine Epix et en général la cohérence des différentes applications Pix.

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr12375.review.pix.fr)
  * [App (.org)](https://app-pr12375.review.pix.org)
  * [Orga (.fr)](https://orga-pr12375.review.pix.fr)
  * [Orga (.org)](https://orga-pr12375.review.pix.org)
  * [Certif (.fr)](https://certif-pr12375.review.pix.fr)
  * [Certif (.org)](https://certif-pr12375.review.pix.org)
  * [Junior](https://junior-pr12375.review.pix.fr)
  * [Admin](https://admin-pr12375.review.pix.fr)
  * [API](https://api-pr12375.review.pix.fr/api/)
  * [API MaDDo](https://pix-api-maddo-review-pr12375.osc-fr1.scalingo.io/api/)
  * [Audit Logger](https://pix-audit-logger-review-pr12375.osc-fr1.scalingo.io/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr12375/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr12375/environment)
  * [scalingo api-maddo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-maddo-review-pr12375/environment)
  * [scalingo audit-logger](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr12375/environment)

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr12375.review.pix.fr)
  * [App (.org)](https://app-pr12375.review.pix.org)
  * [Orga (.fr)](https://orga-pr12375.review.pix.fr)
  * [Orga (.org)](https://orga-pr12375.review.pix.org)
  * [Certif (.fr)](https://certif-pr12375.review.pix.fr)
  * [Certif (.org)](https://certif-pr12375.review.pix.org)
  * [Junior](https://junior-pr12375.review.pix.fr)
  * [Admin](https://admin-pr12375.review.pix.fr)
  * [API](https://api-pr12375.review.pix.fr/api/)
  * [API MaDDo](https://pix-api-maddo-review-pr12375.osc-fr1.scalingo.io/api/)
  * [Audit Logger](https://pix-audit-logger-review-pr12375.osc-fr1.scalingo.io/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr12375/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr12375/environment)
  * [scalingo api-maddo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-maddo-review-pr12375/environment)
  * [scalingo audit-logger](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr12375/environment)

pix-service commented:
:tada: This PR is included in version 5.125.0 :tada:

The release is available on [GitHub release](https://github.com/1024pix/pix/releases/tag/v5.125.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
